### PR TITLE
py-babel: add 2.10.3

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/python-babel/babel/tree/v2.10.3

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed sha256. Defer merge decision to @adamjstewart

File: var/spack/repos/builtin/packages/py-babel/package.py [hunk 1]
```diff
@@ -13,13 +13,16 @@ class PyBabel(PythonPackage):
...
+    git = "https://github.com/python-babel/babel"
 
+    version("2.10.3", sha256="7614553711ee97490f732126dc077f8d0ae084ebc6a96e23db1482afabdb2c51")
     version("2.9.1", sha256="bc0c176f9f6a994582230df350aa6e05ba2ebe4b3ac317eab29d9be5d2768da0")
     version("2.7.0", sha256="e86135ae101e31e2c8ec20a4e0c5220f4eed12487d5cf3f78be7e98d3a57fc28")
     version("2.6.0", sha256="8cba50f48c529ca3fa18cf81fa9403be176d374ac4d60738b839122dfaaa3d23")
     version("2.4.0", sha256="8c98f5e5f8f5f088571f2c6bd88d530e331cbbcb95a7311a0db69d3dca7ec563")
     version("2.3.4", sha256="c535c4403802f6eb38173cd4863e419e2274921a01a8aad8a5b497c131c62875")
 
+    depends_on("python@3.6:", when="@10:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-babel/package.py]:
```suggestion
    depends_on("python@3.6:", when="@2.10:", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-babel/package.py (hunk 1 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-babel/package.py]: Ups... Thanks!

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Pull request merged**